server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/StructureComparisonService.java

```diff
@@ -0,0 +1,254 @@
+
+        log.info("StructureComparisonService create a new structure comparison task entity, id={}",
+                structureComparisonTaskEntity.getId());
+        DBStructureComparisonConfig srcConfig = initSourceComparisonConfig(parameters);
+        DBStructureComparisonConfig tgtConfig = initTargetComparisonConfig(parameters);
+
+        StructureComparisonTask task =
+                new StructureComparisonTask(srcConfig, tgtConfig, taskId, structureComparisonTaskEntity.getId(),
+                        authenticationFacade.currentUser(), structureComparisonTaskRepository,
+                        structureComparisonTaskResultRepository, objectStorageFacade);
+        Future<DBStructureComparisonTaskResult> resultFuture = executor.submit(task);
+
+        return new StructureComparisonContext(task, resultFuture);
+    }
+
+    private DBStructureComparisonConfig initSourceComparisonConfig(DBStructureComparisonParameter parameters) {
+        DBStructureComparisonConfig srcConfig = new DBStructureComparisonConfig();
+
+        DatabaseEntity database = getDatabaseEntityByDatabaseId(parameters.getSourceDatabaseId());
+        ConnectionConfig connectionConfig = getConnectionConfigByDatabaseEntity(database);
+
+        srcConfig.setSchemaName(database.getName());
+        srcConfig.setConnectType(connectionConfig.getType());
+        srcConfig.setDataSource(new DruidDataSourceFactory(connectionConfig).getDataSource());
+        srcConfig.setToComparedObjectTypes(Collections.singleton(DBObjectType.TABLE));
+        if (parameters.getComparisonScope() == ComparisonScope.PART) {
+            Map<DBObjectType, Set<String>> blackListMap = new HashMap<>();
+            blackListMap.put(DBObjectType.TABLE, new HashSet<>(parameters.getTableNamesToBeCompared()));
+            srcConfig.setBlackListMap(blackListMap);
+        }
+
+        return srcConfig;
+    }
+
+    private DBStructureComparisonConfig initTargetComparisonConfig(DBStructureComparisonParameter parameters) {
+        DBStructureComparisonConfig tgtConfig = new DBStructureComparisonConfig();
+
+        DatabaseEntity database = getDatabaseEntityByDatabaseId(parameters.getTargetDatabaseId());
+        ConnectionConfig connectionConfig = getConnectionConfigByDatabaseEntity(database);
+        tgtConfig.setSchemaName(database.getName());
+        tgtConfig.setConnectType(connectionConfig.getType());
+        tgtConfig.setDataSource(new DruidDataSourceFactory(connectionConfig).getDataSource());
+        return tgtConfig;
+    }
+
+    private DatabaseEntity getDatabaseEntityByDatabaseId(Long databaseId) {
+        return databaseRepository.findById(databaseId).orElseThrow(
+                () -> new NotFoundException(ResourceType.ODC_DATABASE, "source database id", databaseId));
+    }
+
+    private ConnectionConfig getConnectionConfigByDatabaseEntity(DatabaseEntity databaseEntity) {
+        Long connectionId = databaseEntity.getConnectionId();
```

Comment:
so, how to verify the user that submit this task can access this connection config?